policy: Allow DNS proxy only on port 53.

jrajahalme opened the pull request:
Forthcoming TPROXY PR limits the DNS Proxy support to port 53. Make
policy validation fail if no port, or other than port 53 is specified
for a toPorts rule with a DNS policy. Add this limit also to the docs.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/7566)
<!-- Reviewable:end -->

raybejjani commented on pkg/policy/api/rule_validation.go:
```diff
@@ -309,6 +309,14 @@ func (pr *L7Rules) sanitize() error {
+		}
+		for _, port := range ports {
+			if port.Port != "53" {
+				return fmt.Errorf("DNS rules are only allowed on port 53")
```
I don't think we have the reason for this written down anywhere. It might be useful to note it here and when it could be changed, but that might be more clear once https://github.com/cilium/cilium/pull/6921 is merged.

jrajahalme replied:
I'll add a comment and fire another test run.

jrajahalme committed:
policy: Allow DNS proxy only on port 53.

Forthcoming TPROXY PR limits the DNS Proxy support to port 53. Make
policy validation fail if no port, or other than port 53 is specified
for a toPorts rule with a DNS policy. Add this limit also to the docs.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>
